torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py

```diff
@@ -268,36 +258,27 @@ def _annotate_conv_relu(
     quantization_config: Optional[QuantizationConfig],
     filter_fn: Optional[Callable[[Node], bool]] = None,
 ) -> Optional[List[List[Node]]]:
-    fused_partitions1 = find_sequential_partitions(
-        gm, [torch.nn.Conv2d, torch.nn.ReLU], filter_fn
-    )
-    fused_partitions2 = find_sequential_partitions(
-        gm, [torch.nn.Conv1d, torch.nn.ReLU], filter_fn
-    )
     annotated_partitions = []
-    for fused_partition in fused_partitions1 + fused_partitions2:
-        conv_partition, relu_partition = fused_partition
-        if len(relu_partition.output_nodes) > 1:
-            raise ValueError("Relu partition has more than one output node")
-        relu_node = relu_partition.output_nodes[0]
-        if len(conv_partition.output_nodes) > 1:
-            raise ValueError("conv partition has more than one output node")
-        conv_node = conv_partition.output_nodes[0]
-
-        if not isinstance(conv_node, Node):
-            raise ValueError(f"{conv_node} is not a Node")
-        if conv_node.op != "call_function" or conv_node.target not in [
-            torch.ops.aten.conv1d.default,
-            torch.ops.aten.conv2d.default,
-        ]:
-            raise ValueError(f"{conv_node} is not an aten conv1d or conv2d operator")
-        if relu_node.op != "call_function" or relu_node.target not in [
+    for n in gm.graph.nodes:
+        if n != "call_function" or n.target not in [
             torch.ops.aten.relu.default,
             torch.ops.aten.relu_.default,
         ]:
-            raise ValueError(f"{relu_node} is not an aten relu operator")
+            continue
+        relu_node = n
+        maybe_conv_node = n.args[0]
+        if not isinstance(maybe_conv_node, Node) or maybe_conv_node != "call_function" or maybe_conv_node.target not in [
+            torch.ops.aten.conv1d.default,
+            torch.ops.aten.conv2d.default,
+        ]:
```

Comment:
I think we should consider refactoring this into either
1. subgraph matcher based OR
2. refactor find_sequential_partitions or add another util there to take in a list of partitions are find those which are sequentially connected. 

The whole list of if conditions is not great.

Reply:
so in general you'll need to get a conv_node and a relu_node, if the API you have did not give a general access to these things, it will still be hacky and making assumptions. let's say `find_connected_nodes([aten.conv2d, aten.relu], [aten.cov1d, aten.relu])`, what API will this function provide for accessing conv2d and relu node? and what if this API takes 10 ops and connected in a graph?

but this is XNNPACKQuantizer so please feel free to make any APIs, it might be better for edge team to design and own these APIs though, would you be OK to land this one first and you can follow up later to add these APIs?

Reply:
> what API will this function provide for accessing conv2d and relu node? and what if this API takes 10 ops and connected in a graph (instead of just sequentially)?

Thats a good point. However you just need input and outptu nodes, right? Anyhow, I do see your point. If we want to do this right, it will have to be something that can handle many cases.

I am fine landing this and we can own and update

Reply:
yeah for most use cases we probably just need input/output, it's fine if you want to start with this and then expand it to more general utils. I was thinking of building this general util actually, but have to figure out the priority.